core/federated/federate.c

```diff
@@ -282,6 +282,7 @@ static void update_last_known_status_on_input_port(environment_t* env, tag_t tag
     // federate that is far ahead of other upstream federates in logical time.
     lf_update_max_level(_fed.last_TAG, _fed.is_last_TAG_provisional);
     lf_cond_broadcast(&lf_port_status_changed);
+    lf_cond_broadcast(&env->event_q_changed);
```

Comment:
Why is this broadcast added here? It doesn't seem like this function should have any effect on the event queue...

Reply:
It is added because there are threads blocking on a change in an empty event queue that will never occur because the federation is shutting down.  In effect, once you know that there will be no more events, this _is_ a change in the status of the event queue.